pilot/pkg/model/jwks_resolver.go

```diff
@@ -224,16 +223,17 @@ func newJwksResolverWithCABundlePaths(
 
 var errEmptyPubKeyFoundInCache = errors.New("empty public key found in cache")
 
-// GetPublicKey returns the  JWT public key if it is available in the cache
+// GetPublicKey returns the JWT public key if it is available in the cache
 // or fetch with from jwksuri if there is a error while fetching then it adds the
 // jwksURI in the cache to fetch the public key in the background process
-func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, error) {
+func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string, timeout time.Duration) (string, error) {
```

Comment:
nit: can you pass ctx

Reply:
Added the ctx variable to GetPublicKey (and to the functions that are called by GetPublicKey() resolveJwksURIUsingOpenID() getRemoteContentWithRetry)